contrib/compiler-rt/lib/sanitizer_common/sanitizer_linux_libcdep.cc

```diff
@@ -349,14 +349,37 @@ static void **ThreadSelfSegbase() {
 # elif defined(__x86_64__)
   // sysarch(AMD64_GET_FSBASE, segbase);
   __asm __volatile("movq %%fs:0, %0" : "=r" (segbase));
+# elif defined(__mips__)
+  // MIPS uses TLS variant I. The thread pointer (in hardware register $29)
+  // points to the end of the TCB + 0x7000.  The TCB contains two pointers,
+  // one to the dtv and the second to the struct pthread.
+  const uptr kTlsTcbOffset = 0x7000;
```

Comment:
I saw this 0x7000 and was confused because libc has an offset of 0x8000.  https://github.com/CTSRD-CHERI/cheribsd/blob/master/lib/libc/gen/tls.c#L108